hoptimator-jdbc/src/main/java/com/linkedin/hoptimator/jdbc/CompatibilityValidatorBase.java

```diff
@@ -0,0 +1,36 @@
+package com.linkedin.hoptimator.jdbc;
+
+import com.linkedin.hoptimator.Validator;
+
+import org.apache.calcite.jdbc.CalciteSchema;
+import org.apache.calcite.schema.Table;
+import org.apache.calcite.schema.SchemaPlus;
+
+/** Base class for shared schema evolution validators.  */
+abstract class CompatibilityValidatorBase implements Validator<SchemaPlus> {
+  
+  @Override
+  public void validate(SchemaPlus schema, Issues issues) {
+    try {
+      CalciteSchema originalSchema = schema.unwrap(CalciteSchema.class);
+      if (originalSchema == null || originalSchema.schema == null) {
+        throw new IllegalArgumentException("Null original schema (BUG)");
+      }
+      for (String x : schema.getTableNames()) {
+        Table table = schema.getTable(x);
+        Table originalTable = originalSchema.schema.getTable(x);
+        if (table == null) {
+          throw new IllegalArgumentException("Null table (BUG)");
+        }
+        if (originalTable == null) {
+          continue;
+        }
+        validate(table, originalTable, issues.child(x));
+      }
+    } catch (ClassCastException e) {
+      // nop
```

Comment:
I see this `ClassCastException` being swallowed in a few spots, why is this?

Reply:
`schema.unwrap(CalciteSchema.class)` will throw a `ClassCastException` if the schema is not a `CalciteSchema`. It's hard to tell if it will throw or not, so I just try and catch. Incidentally, `java.sql.Wrapper` (which, I assume, is what Calcite's `Wrapper` is based on) has a `isWrapperFor(clazz)` method. But I don't think Calcite's has such a test.